File: 백준/Gold/6198. 옥상 정원 꾸미기/옥상 정원 꾸미기.py
import sys

n = int(sys.stdin.readline())
# 수정 1: 각 줄마다 하나씩 입력받아야 함
towers = []
for _ in range(n):
    towers.append(int(sys.stdin.readline()))

stack = []
answer = 0  # 수정 2: result -> answer로 변수명 통일

for i in range(n):
    # 수정 3: 조건을 <= 로 변경 (같거나 낮은 건물은 볼 수 없음)
    while stack and stack[-1][1] <= towers[i]:
        stack.pop()
    
    # 스택에 남아있는 빌딩들은 모두 현재 빌딩을 볼 수 있음
    answer += len(stack)
    
    # 현재 빌딩을 스택에 추가
    stack.append([i, towers[i]])

print(answer)

File: 백준/Gold/6198. 옥상 정원 꾸미기/옥상 정원 꾸미기.js
const fs = require("fs");
const input = fs.readFileSync('/dev/stdin').toString().split('\n');

const n = Number(input[0])
const towers = [];

for (let i = 1; i <= n; i++) towers.push(Number(input[i])); //높이만 

let stack = []; //[인덱스, 높이]
let answer = 0;

for (let i = 0; i < n; i++) {
  //stack 에 대한 검사
  // 그리고 현재 타워보다 낮으면, 빼야함
  while (stack.length > 0 && stack[stack.length - 1][1] <= towers[i]) {
    stack.pop()
  }

  //남아 있는 것들은 현재 타워를 볼 수 있는 것이기 때문에 더하기
  answer += stack.length;

  stack.push([i, towers[i]])
}

console.log(answer);